### PR TITLE
Add Sentry

### PR DESCRIPTION
**Pull request by ryelo**
Default configuration should capture all fatal events in all apps using chrome

If other apps want to roll their own errors, they have the ability to do so.

### CODE REVIEW

File: src/js/sentry.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        environment: `PROD${betaCheck}`,
+        maxBreadcrumbs: 50,
+        attachStacktrace: true,
+        debug: true
+    });
+}
+
+/* eslint-disable camelcase */
+function sentryUser(user) {
+    // TODO: Add request_id to this
```
Comment (kruai) [inline src/js/sentry.js]: Should the request_id be a part of this PR, or is it OK to leave it off for now?

File: src/js/sentry.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import * as Sentry from '@sentry/browser';
+
+const API_KEY = 'https://80e5a70255df4bd3ba6eb3b4bfebc58c@sentry.io/1466891';
```
Comment (kruai) [inline src/js/sentry.js]: Is it OK for this to be in the code?

File: src/js/sentry.js (hunk 1 above)
Comment (ryelo) [reply src/js/sentry.js]: It's being left off for now. This is something to circle back later.

File: src/js/sentry.js (hunk 2 above)
Comment (ryelo) [reply src/js/sentry.js]: Not sure. We're exposing the api key for the pendo stuff in `analytics.js`, so I thought it should be fine? We can always wrap it up and do it somewhere else.

Comment (kruai) [reply src/js/sentry.js]: If it's already exposed, I guess we might as well do the same here. I'm not sure how sensitive the api key is, but if it is, we'll probably want to look at injecting it for both sentry and pendo.

File: src/js/chrome.js [hunk 3]
```diff
@@ -8,7 +9,7 @@ const libjwt = auth();
 function noop () {}
 
 libjwt.initPromise.then(() => {
-    libjwt.jwt.getUserInfo().then(analytics).catch(noop);
+    libjwt.jwt.getUserInfo().then(analytics && sentry).catch(noop);
```
Comment (karelhala) [inline src/js/chrome.js]: I don't think that this is going to work. It's going to run analytics and then dies. To fix this you should do something like
```JS
libjwt.jwt.getUserInfo().then((...data) => {
  analytics(...data);
  sentry(...data);
}).catch(noop);
```
This way any data returned ffrom getUser info will be passed in same way.

File: src/js/sentry.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// - Sentry Options
+// dsn: key
+// environment: logs PROD or PROD Beta for filtering
+// maxBreadcrumbs, if there is an error, trace back up to (x) lines if needed
+// attachStacktrace: attach the actual console logs
+// debug: will attempt to print out useful debugging information if something goes wrong with sending the event
+// sampleRate: 0.0 to 1.0 - percentage of events to send (1.0 by default)
+
+function initSentry() {
+    let betaCheck = (window.location.pathname.split('/')[1] === 'beta' ? ' Beta' : '');
```
Comment (karelhala) [inline src/js/sentry.js]: Nitpick: No need to assign to variable if it's going to be used only in one place.

File: src/js/chrome.js (hunk 3 above)
Comment (ryelo) [reply src/js/chrome.js]: Did you test the code locally? At least for me, it was hitting both analytics and sentry using this method.

Comment (karelhala) [reply src/js/chrome.js]: Oh `analytics` returns true? In that case we are good with shortcutting.

File: src/js/sentry.js (hunk 4 above)
Comment (ryelo) [reply src/js/sentry.js]: @karelhala Sentry won't init with the window function inside the object. You have to get the location before inserting it into the object like it is now.

Comment (karelhala) [reply src/js/sentry.js]: Oh okay.

**Pull request merged**